homeassistant/components/recorder/statistics.py

```diff
@@ -2221,9 +2283,75 @@ def _import_statistics_with_session(
         else:
             _insert_statistics(session, table, metadata_id, stat)
 
+    if table != StatisticsShortTerm:
+        return True
+
+    # We just inserted new short term statistics, so we need to update the
+    # latest_statistics_short_term_ids table that tracks what the newest id is
+    # for the metadata_id.
+    session.execute(
+        _delete_latest_statistics_short_term_ids_by_metadata_ids_stmt({metadata_id}),
+    )
+    if latest := cast(
+        Sequence[Row],
+        execute_stmt_lambda_element(
+            session, _find_latest_short_term_statistic_stmt(metadata_id), orm_rows=False
+        ),
+    ):
+        session.add(
+            LatestStatisticsShortTermIDs(metadata_id=metadata_id, id=latest[0].id)
+        )
+
     return True
 
 
+def _find_latest_short_term_statistic_stmt(
+    metadata_id: int,
+) -> StatementLambdaElement:
+    """Find the latest short term statistics for a list of metadata_ids."""
+    return lambda_stmt(
```

Comment:
We need to add a comment here that we should not optimize this to be a single query for all ids since postgresql will optimize that poorly and do a full scan.

combined query: `(actual time=2.218..893.909 rows=170531 loops=1)`

single query: `(actual time=0.301..0.301 rows=1 loops=1)`

Reply:
comment added in 9de895a348376fe589c4fba5c31794b491b6ce85